independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -57,6 +57,8 @@
 
     private final Map<DotName, ClassInfo> interceptorBindings;
 
+    private final Map<DotName, Set<String>> nonBindingFields;
```

Comment:
@mkouba can you look over these ArC changes?

Reply:
I already did (in Michal's first version of the code and this bit doesn't seem to have changed). In general this is correct and will work. The question is whether we somehow want to weave in annotations transformer directly into the registrar so that we avoid hardcoding non-binding fields and allow others to view that change via annotation store. But I am not quite sure how what would work, WDYT @mkouba ?